File: ai-generator/ai_generator.py
import os
import openai
import sys
from flask import Flask, jsonify, request

class LLM():
    def __init__(self, model="dall-e-3", chat_history=[], configuration_prompt="dalle"):
        # key = os.getenv("OPENAI_API_KEY")
        self.client = openai.OpenAI()
        self.model = model
        self.chat_history = chat_history
        self.configuration_prompt = configuration_prompt

    def ask_gpt(self, user_input: str):
        response = (self.client.images.generate(
            model=self.model,
            prompt=user_input,
            size="1024x1024",
            quality="standard",
            n=1))
        return response.data[0].url

llm = LLM()
app = Flask(__name__)

@app.route('/', methods=['GET'])
def get_prompt():
    try :
        # prompt = "An owl destrying chains"
        json_file = request.get_json()
        print(json_file, file=sys.stderr)
        prompt = json_file['prompt']
        print(prompt, file=sys.stderr)
    except Exception as e:
        return jsonify({"error": "404"})
    msg = llm.ask_gpt(prompt)
    print(jsonify({"url": msg}), file=sys.stderr)
    return jsonify({"url": msg})
    # respond = requests.post("http://localhost:5000/openai/images/generate", json={"url": msg})
    # return jsonify(respond.json())

if __name__ == "__main__":
    app.run()

File: backend/openai/ai_generator.js
const express = require('express');
const bodyParser = require('body-parser');
const dotenv = require("dotenv");
const cors = require('cors'); // Import CORS middleware
const axios = require('axios'); // Import axios for making HTTP requests
const bodyparser = require('body-parser');
const OpenAI = require("openai")

dotenv.config()

const app = express();
app.use(bodyparser.json());
const port = 5001;

const openai = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY
});

app.use(cors());

app.post('/openai', async (req, res) => {
  const resu = await openai.images.generate({
    model: "dall-e-3",
    prompt: req.body.prompt,
    n: 1,
    size: '1024x1024'
  })
  const url = resu.data[0].url;
  console.log(url)

  res.status(200).json({
    success: true,
    data: url
  });
});

app.use(express.static('public'));
app.listen(port, () => {
  console.log(`Server running on http://localhost:${port}`);
});
